datafusion/functions-array/Cargo.toml

```diff
@@ -52,6 +52,7 @@ datafusion-functions = { workspace = true }
 itertools = { version = "0.12", features = ["use_std"] }
 log = { workspace = true }
 paste = "1.0.14"
+sqlparser = { workspace = true }
```

Comment:
Since datafusion-expr already has a explicit dependency on `sqlparser` -- https://github.com/apache/datafusion/blob/d19487c9682738066ff3394d7492943adaadb509/datafusion/expr/Cargo.toml#L49

I think it would be cleaner here to avoid the explicit dependency on `sqlparser` and instead export it in datafusion-expr.

SO like in `datafusion/expr/src/lib.rs` put a `pub use sqlparser`;

That would ensure the versions always matched (though I think workspace mostly takes care of this) as well as make it clear that the `functions-array` really depends on the API in datafusion-expr rather than sqlparser directly

Reply:
btw do we also need to cleanup `datafusion-sql` as well as it import datafusion-expr and sqlparser too